src/client-utils/file-hashing.ts

```diff
@@ -0,0 +1,39 @@
+import jsSHA from 'jssha';
+
+const computeSHA256 = async (file: File) => {
+  // Read the file as an ArrayBuffer
+  const fileBuffer = await file.arrayBuffer();
+
+  // Compute the hash
+  const hashArrayBuffer = await crypto.subtle.digest('SHA-256', fileBuffer);
+
+  // Convert hash to hexadecimal
+  const hashArray = Array.from(new Uint8Array(hashArrayBuffer));
+  const hashHex = hashArray.map((byte) => byte.toString(16).padStart(2, '0')).join('');
+
+  return hashHex;
+};
+
+const computeSHA256jsSHA = async (file: File) => {
+  const reader = file.stream().getReader({ mode: 'byob' });
+  const shaObj = new jsSHA('SHA-256', 'ARRAYBUFFER');
+
+  const chunkSize = 100 * 1024 * 1024; // 100MB
```

Comment:
Seems like both of these could be declared outside so they're not instantiated every time this function gets called